ibis-server/tests/routers/v2/connector/test_postgres.py

```diff
@@ -724,19 +726,93 @@ async def test_model_substitute(client, manifest_str, postgres: PostgresContaine
         == '"SELECT * FROM \\"my_catalog\\".\\"my_schema\\".\\"Orders\\" AS \\"orders\\""'
     )
 
+    # Test without catalog and schema in SQL but in headers(x-user-xxx)
+    response = await client.post(
+        url=f"{base_url}/model-substitute",
+        headers={"x-user-catalog": "test", "x-user-schema": "public"},
+        json={
+            "connectionInfo": connection_info,
+            "manifestStr": manifest_str,
+            "sql": 'SELECT * FROM "orders"',
+        },
+    )
+    assert response.status_code == 200
+    assert (
+        response.text
+        == '"SELECT * FROM \\"my_catalog\\".\\"my_schema\\".\\"Orders\\" AS \\"orders\\""'
+    )
+
+    # Test only have x-user-catalog
+    response = await client.post(
+        url=f"{base_url}/model-substitute",
+        headers={"x-user-catalog": "test"},
+        json={
+            "connectionInfo": connection_info,
+            "manifestStr": manifest_str,
+            "sql": 'SELECT * FROM "orders"',
+        },
+    )
+    assert response.status_code == 422
+
+    # Test only have x-user-catalog but have schema in SQL
+    response = await client.post(
+        url=f"{base_url}/model-substitute",
+        headers={"x-user-catalog": "test"},
+        json={
+            "connectionInfo": connection_info,
+            "manifestStr": manifest_str,
+            "sql": 'SELECT * FROM "public"."orders"',
+        },
+    )
+    assert response.status_code == 422
```

Comment:
Why does this case fail? If the default catalog is `test`, the table name will be `test.public.orders`. I think it can map to `"my_catalog"."my_schema"."Orders"`, right?